alcs-frontend/src/app/features/application/overview/overview.component.ts

```diff
@@ -162,61 +160,68 @@ export class OverviewComponent implements OnInit, OnDestroy {
     decisions: ApplicationDecisionDto[],
     reconsiderations: ApplicationReconsiderationDto[],
     modifications: ApplicationModificationDto[],
-    applicationReview: ApplicationReviewDto | undefined
+    statusHistory: ApplicationSubmissionToSubmissionStatusDto[]
   ): TimelineEvent[] {
     const mappedEvents: TimelineEvent[] = [];
-    if (application.dateSubmittedToAlc) {
-      mappedEvents.push({
-        name:
-          applicationReview && applicationReview.isAuthorized === false
-            ? 'L/FNG Refused to Forward'
-            : 'Submitted to ALC',
-        startDate: new Date(application.dateSubmittedToAlc + SORTING_ORDER.SUBMITTED),
-        isFulfilled: true,
-      });
-    }
 
-    if (application.dateAcknowledgedIncomplete) {
-      mappedEvents.push({
-        name: 'Acknowledged Incomplete',
-        startDate: new Date(application.dateAcknowledgedIncomplete + SORTING_ORDER.ACKNOWLEDGED_INCOMPLETE),
-        isFulfilled: true,
-      });
+    const statusesToInclude = statusHistory.filter(
+      (status) => ![SUBMISSION_STATUS.IN_REVIEW_BY_ALC].includes(status.status.code)
+    );
+    for (const status of statusesToInclude) {
+      if (status.effectiveDate) {
+        let htmlText = `<strong>${status.status.label}</strong>`;
+
+        if (status.status.code === SUBMISSION_STATUS.RECEIVED_BY_ALC) {
+          htmlText = 'Received All Items - <strong>Received by ALC</strong>';
+        }
+
+        if (status.status.code === SUBMISSION_STATUS.IN_PROGRESS) {
+          htmlText = 'Created - <strong>In Progress</strong>';
+        }
+
+        if (status.status.code === SUBMISSION_STATUS.SUBMITTED_TO_ALC_INCOMPLETE) {
+          htmlText = 'Acknowledge Incomplete - <strong>Submitted to ALC - Incomplete</strong>';
+        }
+
+        mappedEvents.push({
+          htmlText,
+          startDate: new Date(status.effectiveDate + status.status.weight),
```

Comment:
Is the weight coming from backend and similar concept as SORTING_ORDER?

Reply:
Yes! This is an easy way for me to sort them in the same order that the backend applies.